pkg/console/subresource/route/route.go

```diff
@@ -162,31 +166,42 @@ func wildcard() routev1.WildcardPolicyType {
 	return routev1.WildcardPolicyNone
 }
 
+// The canonical host is the ingress on the route with a host that matches the
+// ingressConfig.Spec.Domain.  When we have an ingress that matches, and when it
+// is admitted, the console will be fully accessible.
+func GetCanonicalHost(route *routev1.Route, ingressConfig *configv1.Ingress) (string, error) {
```

Comment:
Let's add tests for this one :)

Reply:
Absolutely, if this seems the right approach, I'll def add tests for them.